libsolidity/ast/Types.cpp

```diff
@@ -2000,8 +2000,7 @@ TypePointer TupleType::closestTemporaryType(TypePointer const& _targetType) cons
 
 FunctionType::FunctionType(FunctionDefinition const& _function, bool _isInternal):
 	m_kind(_isInternal ? Kind::Internal : Kind::External),
-	m_isConstant(_function.isDeclaredConst()),
-	m_isPayable(_isInternal ? false : _function.isPayable()),
+	m_stateMutability(_function.stateMutability()),
```

Comment:
no ,if it was `view` it should stay view.

`_isInternal && _function.stateMutability.isPayable() ? StateMutability::NonPayable : _function.stateMutability.isPayable()`.